zerver/logging_handlers.py

```diff
@@ -122,8 +125,8 @@ def emit(self, record: logging.LogRecord) -> None:
             report['log_module'] = find_log_caller_module(record)
             report['log_lineno'] = record.lineno
 
-            if hasattr(record, "request"):
-                add_request_metadata(report, record.request)  # type: ignore[attr-defined]  # record.request is added dynamically
+            if record in record_request:
+                add_request_metadata(report, record_request[record])
```

Comment:
It’s possible I’m missing the purpose of `record.request` here—it seems to only be set in the test suite?

Reply:
Oh I found it, will leave this out for now.